### ISSUE TITLE
request for enhancement

### ISSUE THREAD

**Issue by visualperception** [GuardRex/vscode-status-bar-tasks#10]
vscode varibales in tasks.json provides for various variables. The one I'm looking at specifically is:
${workspaceRoot}

This returns ( I think) the root folder as opened with vscode file menu. i.e. is the root of your editing environment and what is returned is the full file path to folder. e.g. c:\folder1\folder2\folder3 if that happened to be your folder structure to the workspaceRoot

For me when working on a website, this is my website root folder (and I have many different website root folders and not just one). My local apache test server is setup to run websites from these root folders so I have urls which are following for each website:

http://localhost/rootfolder

What I would like access to is a variable which just returns the name of workspaceRoot and not its file path. So from example above I would like access to a variable which returns just:

folder3

not sure how you could implement that but as a suggestion could a false variable be setup called say ${workspaceRootName} and if that is in any of the arguments then the last foldername is stripped from ${workspaceRoot} and ${workspaceRootName} be replaced with it in the argument?

So if my tasks.json code was:

```
{
    // See https://go.microsoft.com/fwlink/?LinkId=733558
    // for the documentation about the tasks.json format
    "version": "0.1.0",
    "command": "cmd",
    "isShellCommand": true,
    "showOutput": "always",
    "args": ["/C"],
    "tasks": [
        {
            "taskName": "iexplore",
            "suppressTaskName": true,
            "args" : ["iexplore.exe localhost/${workspaceRootName}${relativeFile}"],
            "showOutput": "always"
        }
             ]
} 

```

Then I would get:

localhost/folder3/relative-path-to-file 

as the url passed to my browser.

I ask for this becasue vscode does not provide the name as just folder name. It always return path with drive and that is not good for website urls and the full path is not by default the correct path for a website anyway depending on how your webserver is configured.

I have tried 3 other extensions which claim to run current open file in a browser but all 3 of them lock up the browser with vscode extension still connected in some way. This seems to happen with wordpress site and could be htaccess redirests related but I don't know.
So looking at your extension and having done some testing that site doesn't get locked up, it would seem to work fine except for ability to generate correct url automatically. (and bug that I posted previously)

It would be nice if vscode had its own native variable for root folder name but as far as I've found out so far it doesn't and json doesn't handle regex to strip it out.

Thanks

**Comment by guardrex** [comment]
Let me take a look at this as soon as I can get back to it. I'm getting slammed with professional work right now ... I'm on a 10-12 hour day 7-days a week trying to catch up.

**Comment by visualperception** [comment]
as it happens I have now posted a request for a new system variable which returns the workspaceRootFolderName without its path. This is in progress and will negate the need to use an extension to build a url since in task args I will just be able to put ${workspaceRootFolderName}/$relativeFile and pass that directly to iexplore in my task (and also assign a key to the task without using an extension.
So that will do what I need. So strictly speaking I don't need these changes now but the bug that wasn't a bug will also need to handle the new system variable ${workspaceRootFolderName} which is awating merge to VScode Master branch in September (probably end of month. Its already been coded)

Thanks

p.s. this new system variable should make a few extensions redundant since they existed largely to get around the fact that you can't edit file paths in json to create urls. Your extension is a bit different in that it displays tasks for clicking on but I guess all extensions need to be kept uptodate with core vscode and system variables if people are likely to be using them.

**Comment by guardrex** [comment]
> kept uptodate with core vscode and system variables if people are likely to be using them.

I agree with that sentiment. I do need to try and dedicate more time to keeping this mildly popular extension up-to-date. I'm working on a major professional project at the moment that's taking like 12 hours a day and 7 days a week. I think I'll get this project behind me in about 2+ weeks. I'll take a day at that point and just work the extension forward on the last few feature requests in the issues.

I'll ping back in a few weeks when I get time to address all of the recent feature requests for the extension.

**Comment by guardrex** [comment]
ugh .... 2 weeks ... more like 2+ *months* ... BUT ... licked it!

The string will be ~`${workspaceRootname}`~ `${workspaceRootFolderName}` ,,,

`${workspaceRoot}` | `${workspaceRootFolderName}`
--- | ---
c:\_APPS\folder1 | folder1
c:\_APPS\folder1\folder2 | folder2

This will be implemented in the next update (v.0.1.1), which will be out within a day. I'm just waiting on VSC devs to get back to me on another issue/feature that I'm working on. Once I hear back from them and get that other feature implemented, I'll get v0.1.1 up on the Extensions Gallery.

**Comment by guardrex** [comment]
Actually, they added it to VSC as `${workspaceRootFolderName} `, so I'll use the same variable name for the extension.

**Comment by guardrex** [comment]
This is fixed with latest version in the gallery (v.0.1.2). Thanks for pointing out that they added some new vars.
